### DISCUSSION TITLE
Docker build

### DISCUSSION THREAD

**Discussion by garyw-xlabs** [Azure/azure-dev#4316] [Q&A]
hi all.
is there a way to pass a secret to docker when building images,
we have a private nuget package which needs the auth token sending to docker to update the nuget.config file before building the project.
i cant see how i can get this in with azd build.

**Comment by vhvb1989** [comment]
Hello @garyw-xlabs , there's not support in azd by itself for docker-build-secrets.  
Azd supports passing build secrets and args for .Net Aspire projects;  Would that work for you?  or are you using azd with a list of docker-projects? or how's your project?

**Reply by garyw-xlabs** [reply]
hi @vhvb1989 thanks for your response.  we are using .Net Aspire locally but I'm not sure it can help us in the deployment as we need the  ability to deploy each service separately, and to deploy into blue/green revisions in each container app, because of this, and other organisation requirements for where things need to be deployed, we cant use `azd deploy`.   
Does `azd package` create the docker images for each of the services? i cant see them in the logs.  if it does and i can pre determine the name:tag, like we can in a dockerised container app definition in `azure.yaml` then i would be able to use a post package hook to push the images to our company container registry and then continue with out current deployment process.

**Reply by vhvb1989** [reply]
unfortunately, `azd package` is a `no-op` for .Net Aspire.
